ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/results/EngineGetPayloadResultV6110.java

```diff
@@ -0,0 +1,195 @@
+/*
+ * Copyright Hyperledger Besu Contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.ethereum.api.jsonrpc.internal.results;
+
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.parameters.DepositParameter;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.parameters.WithdrawalParameter;
+import org.hyperledger.besu.ethereum.core.BlockHeader;
+import org.hyperledger.besu.ethereum.core.Deposit;
+import org.hyperledger.besu.ethereum.core.Withdrawal;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import com.fasterxml.jackson.annotation.JsonGetter;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonPropertyOrder;
+import org.apache.tuweni.bytes.Bytes32;
+
+@JsonPropertyOrder({
+  "executionPayload",
+  "blockValue",
+})
+public class EngineGetPayloadResultV6110 {
```

Comment:
We might not need a new class for this since you could technically use EngineGetPayloadResultV2 and add optional deposits to that (which would always be empty for shanghai fork for example).

The problem then is EngineGetPayloadResultV2 is a bad name since it really represents EngineGetPayloadResultV2+. I need to think about this :)

The only reason we needed EngineGetPayloadResultV2 to distinguish from EngineGetPayloadResultV1 was due to the change in top-level json structure with the addition of blockValue.

Reply:
Yea I was thinking about this as well. The `execution-apis` repo says the name of the payload will be `ExecutionPayloadV6110`: https://github.com/mkalinin/execution-apis/blob/eip6110/src/engine/experimental/eip6110.md#executionpayloadv6110. Of course from the engineering perspective we don’t have to follow the spec 100%

Reply:
Indeed, ExecutionPayload and EngineGetPayloadResult are not quite equivalent since it's the outer wrapper, but I can understand the association. 
I would have gone with a similar name if I wasn't already thinking about how to remove the version names from these classes entirely :)
This name is fine for now anyway.